CoreDataAccessView/src/au/gov/asd/tac/constellation/views/dataaccess/plugins/experimental/TestParametersPlugin.java

```diff
@@ -178,10 +178,10 @@ public PluginParameters createParameters() {
         selected.setDescription("Only use selected elements");
         params.addParameter(selected);
 
-        final PluginParameter<StringParameterValue> queryName = StringParameterType.build(CoreGlobalParameters.QUERY_NAME_PARAMETER_ID);
+        final PluginParameter<StringParameterValue> queryName = CoreGlobalParameters.getStringParameterType(CoreGlobalParameters.QUERY_NAME_PARAMETER_ID);
```

Comment:
Rather than having the parameter id's as keys (and having to deal with the issue of figuring out what type of parameter it is), couldn't you just build the parameters upfront and make the actual parameter objects the keys?

Reply:
Is that approach scalable? Could you elaborate?

Reply:
> What would the "value" be if the object was the "key"?

@cygnus-x-1 this is what I was going to ask too. Please elaborate when you can.

Reply:
Key was probably the wrong word to use there. If the parameters are built up-front, you could refer to them directly (ie. `CoreGlobalParameters.QUERY_NAME_PARAMETER`), or look them up using their parameter ID. I guess this technically wouldn't make them singletons - however the root issue here was having multiple instances of them built across multiple modules, and that would be addressed. It would also make using them simpler as the developer wouldn't have to know what type they are up-front.